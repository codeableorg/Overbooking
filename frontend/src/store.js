import { createStore, applyMiddleware, compose } from "redux";
import reducer from "./reducer";
import logger from "redux-logger";
import thunk from "redux-thunk";

const composer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(reducer, composer(applyMiddleware(thunk, logger)));

export default store;
