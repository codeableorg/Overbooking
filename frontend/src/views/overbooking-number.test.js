import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import store from "../store";

import OverbookingNumber from "./overbooking-number";

test("CriticalRatio component", () => {
  const { asFragment } = render(
    <Provider store={store}>
      <OverbookingNumber />
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});
