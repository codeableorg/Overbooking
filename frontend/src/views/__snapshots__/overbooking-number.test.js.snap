// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CriticalRatio component 1`] = `
<DocumentFragment>
  <section
    class="css-1bdv45e-OverbookingNumber"
  >
    <img
      alt="airplane"
      class="css-19bmys6-OverbookingNumber"
      src="airplane.png"
    />
    <p
      class="css-1j7e321-OverbookingNumber"
    >
      Critical Ratio: 0.4
    </p>
    <p
      class="css-1j7e321-OverbookingNumber"
    >
      Suggested Overbooking: 3
    </p>
    <p
      class="css-1b4lnb1-OverbookingNumber"
    >
      Total Revenue: $ 30000
    </p>
    <form
      class="css-1utjrzx-OverbookingNumber"
    >
      <span
        class="css-11iy7qm-OverbookingNumber"
        id="rs-bullet"
      >
        0
      </span>
      <input
        class="css-rblfwm-OverbookingNumber"
        id="rs-range-line"
        max="75"
        min="0"
        type="range"
        value="0"
      />
      <p
        class="css-urqw8c-OverbookingNumber"
      >
        Choose your overbooking
      </p>
      <button
        class="css-r4z6bh-OverbookingNumber"
        type="submit"
      >
        Set overbooking
      </button>
    </form>
  </section>
</DocumentFragment>
`;
